crates/eframe/src/native/glow_integration.rs

```diff
@@ -980,8 +980,7 @@ impl GlutinWindowContext {
         if let Some(window) = &window {
             viewport_from_window.insert(window.id(), ViewportId::ROOT);
             window_from_viewport.insert(ViewportId::ROOT, window.id());
-            info.minimized = window.is_minimized();
-            info.maximized = Some(window.is_maximized());
+            egui_winit::update_viewport_info(&mut info, egui_ctx, window);
```

Comment:
`update_viewport_info` doesn't assign `minimized` and `maximized` on Mac (because it's not always safe to do so without a deadlock).

Maybe we could add a `bool` argument to `update_viewport_info` indicating wether or not it is safe to read `minimized` and `maximized` on mac? (it is safe to do during startup, but not from within the even loop)